### PR TITLE
[MM-46382] Make convert to private playbook more prominent

### PR DESCRIPTION
**Pull request by cannalee90**
## Summary

- Add convert to playbook private menu in the dropdown
- Create custom hook for converting playbook private 
- Apply custom hook for playbook access modal for converting private

## Ticket Link

Fixes: https://github.com/mattermost/mattermost-server/issues/21184
JIRA: https://mattermost.atlassian.net/browse/MM-46382

## Checklist
- [x] Telemetry updated
- [x] Unit tests updated(everything except formatted_duration.test.tsx because of i18n issue)

### CODE REVIEW

**Comment by cannalee90** [comment]
gonna fix it asap

File: webapp/src/components/backstage/convert_private_playbook_modal.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import ConfirmModal from 'src/components/widgets/confirmation_modal';
+import {useEditPlaybook} from 'src/hooks';
+
+type ConfirmPlaybookConvertPrivateReturn = [React.ReactNode, (show: boolean) => void];
+type Props = {
+    playbookId: string,
+    refetch?: () => void | undefined,
+}
+
+const useConfirmPlaybookConvertPrivateModal = ({playbookId, refetch}: Props): ConfirmPlaybookConvertPrivateReturn => {
```
Comment (crspeller) [inline webapp/src/components/backstage/convert_private_playbook_modal.tsx:13]: I like that you pulled this out from the access modal. Could we update the usage there?

File: webapp/src/components/backstage/playbook_editor/controls.tsx [hunk 2]
```diff
@@ -440,6 +445,18 @@ const TitleMenuImpl = ({playbook, children, className, editTitle, refetch}: Titl
                     <ExportVariantIcon size={18}/>
                     <FormattedMessage defaultMessage='Export'/>
                 </DropdownMenuItemStyled>
+                {isEligibleToMakePrivate && (
+                    <DropdownMenuItemStyled
+                        role={'button'}
+                        css={`${iconSplitStyling}`}
+                        onClick={() => {
+                            telemetryEventForPlaybook(playbook.id, 'playbook_makeprivate');
```
Comment (crspeller) [inline webapp/src/components/backstage/playbook_editor/controls.tsx:453]: I think we forgot to actually open the modal.

File: webapp/src/components/backstage/playbook_editor/controls.tsx [hunk 3]
```diff
@@ -440,6 +445,18 @@ const TitleMenuImpl = ({playbook, children, className, editTitle, refetch}: Titl
...
+                        }}
+                    >
+                        <FormattedMessage defaultMessage='Convert to private playbook'/>
```
Comment (crspeller) [inline webapp/src/components/backstage/playbook_editor/controls.tsx:456]: We probably want an icon here. @abhijit-singh Can you provide an icon recommendation?

**Review by crspeller — CHANGES_REQUESTED** [review]
Thanks for the contribution!

File: webapp/src/components/backstage/convert_private_playbook_modal.tsx (hunk 1 above)
Comment (cannalee90) [reply webapp/src/components/backstage/convert_private_playbook_modal.tsx:13]: So are you suggesting to make change to use this hook  for access modal?

File: webapp/src/components/backstage/playbook_editor/controls.tsx (hunk 2 above)
Comment (cannalee90) [reply webapp/src/components/backstage/playbook_editor/controls.tsx:453]: I just fix it

File: webapp/src/components/backstage/convert_private_playbook_modal.tsx (hunk 1 above)
Comment (crspeller) [reply webapp/src/components/backstage/convert_private_playbook_modal.tsx:13]: Exactly, so we are using the same code in both places.

File: webapp/src/components/backstage/playbook_editor/controls.tsx (hunk 3 above)
Comment (abhijit-singh) [reply webapp/src/components/backstage/playbook_editor/controls.tsx:457]: The lock-outline icon should work here 
image

Comment (cannalee90) [reply webapp/src/components/backstage/playbook_editor/controls.tsx:459]: I added icon-lock-outline icon!

**Comment by cannalee90** [comment]
@crspeller Hi. As you suggested I just apply my custom hook for playbook access modal but there is a problem. If I convert it to private access modal, the private link I just clicked not disappeared even though lock-icon appeared in background.

The reason why it happed it that the playbook object that used in access-modal is not same as the one that I used in the custom hook. so I just passed the updater function to solve this problem. If you have any other idea please give me a hint for that

Thank you!

**Review by crspeller — APPROVED** [review]
@cannalee90 That's fine for now. Thanks!

**Review by lieut-data — APPROVED** [review]
Super slick! Thanks, @cannalee90 :)

**Review by abhijit-singh — APPROVED** [review]
Looks good, thanks @cannalee90!

**Comment by jordanafung** [comment]
Hi @cannalee90 
You were selected as one of our Top Contributors for Hacktoberfest!
Please fill out [this form](https://docs.google.com/forms/d/11pLB3pF1CmOslYqHr7c7bxFWU6QAEAJuiaKgHD1vJtg) to select your swag as thanks!
Thank you for all you do and for making Mattermost even better!

**Pull request merged**